tests/registry/registry.go

```diff
@@ -6,37 +6,63 @@ package registry
 import (
 	"github.com/onsi/ginkgo/v2"
 
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/genesis"
 	"github.com/ava-labs/hypersdk/tests/workload"
 )
 
-type TestFunc func(t ginkgo.FullGinkgoTInterface, tn workload.TestNetwork)
+type TestFunc func(t ginkgo.FullGinkgoTInterface, tn workload.TestNetwork, authFactories ...chain.AuthFactory)
 
 type namedTest struct {
-	Fnc  TestFunc
-	Name string
+	Fnc               TestFunc
+	Name              string
+	AuthFactories     []chain.AuthFactory
+	requestedBalances []uint64
 }
 type Registry struct {
-	tests []namedTest
+	tests []*namedTest
 }
 
-func (r *Registry) Add(name string, f TestFunc) {
-	r.tests = append(r.tests, namedTest{Fnc: f, Name: name})
+func (r *Registry) Add(name string, f TestFunc, requestedBalances ...uint64) {
+	r.tests = append(r.tests, &namedTest{Fnc: f, Name: name, requestedBalances: requestedBalances})
 }
 
-func (r *Registry) List() []namedTest {
+func (r *Registry) List() []*namedTest {
 	if r == nil {
-		return []namedTest{}
+		return []*namedTest{}
 	}
 	return r.tests
 }
 
+func (r *Registry) GenerateCustomAllocations(generateAuthFactory func() (chain.AuthFactory, error)) ([]*genesis.CustomAllocation, error) {
```

Comment:
The authFactories creation is delayed because when registering the tests, we don't know which private key type to use.